### ISSUE TITLE
Move connection identifiers in interchain accounts port ID to version

### ISSUE THREAD

**Issue by colin-axner** [cosmos/ibc-go#615]
## Summary

Move the connection ID and counterparty connection ID numbers from the portID to the channel version.

## Problem Definition

We recently discussed the possibility of creating ORDERED channels which do not close on packet timeouts. This makes channel recovery in interchain accounts a lot less necessary. If we decide to pursue this change in the future, changing the port ID format will be very difficult, thus it seems desirable to move these fields, which may not be necessary in the future, to the version which can be changed. 

____

#### For Admin Use

- [ ] Not duplicate issue
- [ ] Appropriate labels applied
- [ ] Appropriate contributors tagged/assigned

**Comment by damiannolan** [comment]
I think if we are going with this we should probably JSON encode the payload into a typed `Metadata` field or something along those lines. What do you think?

**Comment by colin-axner** [comment]
As in create a type for the version and JSON encode it? Seems reasonable to me. We won't have to do string parsing, client side representation of the version will be quite ugly I think

**Comment by damiannolan** [comment]
Yeah exactly, what we were talking about before. I think with adding more fields to the version parsing could just get out of hand and JSON encoding would be somewhat cleaner.

Do you mean ugly with the JSON encoding or ugly with extra fields? I think it would be ugly either way I guess 🤣

**Comment by colin-axner** [comment]
How will the JSON encoding look when printed as a string? I guess it might look fine? 

I think using a type is probably a good idea in this instance

**Comment by damiannolan** [comment]
I think it would look fine as long as it's unescaped when used for things such as logging etc
https://go.dev/play/p/eL70Ra3Ia0q

**Comment by gjermundgaraba** [comment]
This issue was mistakenly reopened during some integration work, so closing again now.
